crates/starlight/src/jsrt/object.rs

```diff
@@ -142,6 +144,79 @@ pub fn has_own_property(vm: &mut Runtime, args: &Arguments) -> Result<JsValue, J
     Ok(JsValue::new(obj.get_own_property(vm, prop).is_some()))
 }
 
+pub fn object_get_own_property_descriptor(vm: &mut Runtime, args: &Arguments) -> Result<JsValue, JsValue> {
+    let stack = vm.shadowstack();
+    if args.size() < 2 {
+        return Ok(JsValue::new(Undefined));
+    }
+    let first = args.at(0);
+    let prop = args.at(1);
+    if first.is_jsobject() {
+        letroot!(obj = stack, first.get_jsobject());
+        let name = prop.to_symbol(vm)?;
+
+        match obj.get_own_property(vm, name) {
+            Some(property_descriptor) => {
+                let mut res = JsObject::new_empty(vm);
```

Comment:
this one should be `letroot!(res = stack, JsObject::new_empty(vm));` since GC might be triggered in `define_own_property` later.

Reply:
Oh, that make sense